File: PrimaMateria/PrimaMateria/Classes/DividerLabel.m
//
//  DividerLabel.m
//  PrimaMateria
//
//  Created by Jerry Porter on 1/5/2011.
//  Copyright (c) 2016 xTrensa. All rights reserved.
//

@implementation DividerLabel

- (void)createLayer {    
	self.layer.masksToBounds = YES;
	[self setupColors];
}

- (void)setupColors {
	self.textColor = UIColor.whiteColor;
	[self setHigh_Color: UIColor.lightGrayColor];
	[self setLow_Color: UIColor.grayColor];
}

- (void)awakeFromNib {
	[super awakeFromNib];
	[self createLayer];
}

- (void)setHigh_Color: (UIColor*)color {
	self.highColor = color;
	[self.layer setNeedsDisplay];
}

- (void)setLow_Color: (UIColor*)color {
	self.lowColor = color;
	[self.layer setNeedsDisplay];
}

#pragma mark - Memory Management Methods

@end

File: PrimaMateria/PrimaMateria/Classes/DividerLabel.swift
//
//  DividerLabel.swift
//  PrimaMateria
//
//  Created by Jerry Porter on 4/14/16.
//  Copyright © 2016 xTrensa. All rights reserved.
//

@objc class DividerLabel : UILabel {
    var highColor : UIColor?
    var lowColor : UIColor?
    
    // MARK: - Initialization Methods
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder:aDecoder)!
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.createLayer()
    }
    
    // MARK: - Internal Methods
    
    func createLayer() {
        self.layer.masksToBounds = true
        self.setupColors()
    }
    
    func setupColors() {
        self.textColor = UIColor.whiteColor()
        self.setHigh_Color(UIColor.lightGrayColor())
        self.setLow_Color(UIColor.grayColor())
    }
    
    func setHigh_Color(color: UIColor) {
        self.highColor = color
        self.layer.setNeedsDisplay()
    }
    
    func setLow_Color(color: UIColor) {
        self.lowColor = color
        self.layer.setNeedsDisplay()
    }
}
